webdav/dandidav.py

```diff
@@ -0,0 +1,570 @@
+
+
+class DandiProvider(DAVProvider):
+    def __init__(self) -> None:
+        super().__init__()
+        self.client = DandiAPIClient.for_dandi_instance(INSTANCE, token=TOKEN)
+
+    def get_resource_inst(self, path: str, environ: dict) -> RootCollection:
+        return RootCollection("/", environ).resolve("/", path)
+
+    def is_readonly(self) -> bool:
+        return True
+
+
+class RootCollection(DAVCollection):
+    def get_member_names(self) -> list[str]:
+        return ["dandisets"]
+
+    def get_member(self, name: str) -> DandisetCollection | None:
+        if name == "dandisets":
+            return DandisetCollection(join_uri(self.path, name), self.environ)
+        else:
+            return None
+
+    def is_link(self) -> bool:
+        # Fix for <https://github.com/mar10/wsgidav/issues/301>
+        return False
+
+
+class DandisetCollection(DAVCollection):
+    """Collection of Dandisets in instance"""
+
+    def __init__(self, path: str, environ: dict) -> None:
+        super().__init__(path, environ)
+        self.client: DandiAPIClient = self.provider.client
+
+    def get_member_list(self) -> list[DandisetResource]:
+        return [
+            DandisetResource(join_uri(self.path, d.identifier), self.environ, d)
+            for d in self.client.get_dandisets()
+        ]
+
+    def get_member_names(self) -> list[str]:
+        return [d.identifier for d in self.client.get_dandisets()]
+
+    def get_member(self, name: str) -> DandisetResource | None:
+        if not re.fullmatch(DANDISET_ID_REGEX, name):
+            return None
+        try:
+            d = self.client.get_dandiset(name, lazy=False)
+        except NotFoundError:
+            return None
+        else:
+            return DandisetResource(join_uri(self.path, name), self.environ, d)
+
+    def is_link(self) -> bool:
+        # Fix for <https://github.com/mar10/wsgidav/issues/301>
+        return False
+
+
+class DandisetResource(DAVCollection):
+    def __init__(self, path: str, environ: dict, dandiset: RemoteDandiset) -> None:
+        super().__init__(path, environ)
+        self.dandiset = dandiset
+
+    def get_display_info(self) -> dict[str, str]:
+        return {"type": "Dandiset"}
+
+    def get_member_names(self) -> list[str]:
+        names = ["draft"]
+        if self.dandiset.most_recent_published_version is not None:
+            names.append("latest")
+            names.append("releases")
+        return names
+
+    def get_member(self, name: str) -> VersionResource | ReleasesCollection:
+        if name == "draft":
+            d = self.dandiset.for_version(self.dandiset.draft_version)
+            return VersionResource(join_uri(self.path, name), self.environ, d)
+        elif (
+            name == "latest"
+            and (v := self.dandiset.most_recent_published_version) is not None
+        ):
+            d = self.dandiset.for_version(v)
+            return VersionResource(join_uri(self.path, name), self.environ, d)
```

Comment:
for draft and latest in the listing of dandisets endpoint it also provides sizes -- could sizes be exposed here "for free" without any extra API call?

Reply:
wsgidav doesn't support sizes for collections.

Reply:
what about `quota-used-bytes` as at http://www.webdav.org/specs/rfc4331.html#example-propfind and seems supported in wsgidav: https://github.com/mar10/wsgidav/blob/HEAD/wsgidav/dav_provider.py#L554

Reply:
@yarikoptic That is intended for reporting how much of some storage quota is taken up by the resource and others "similar to" it.  Also, if your goal is for these sizes to show up in the web view, defining `get_used_bytes()` won't accomplish that.